rasa/core/brokers/pika.py

```diff
@@ -169,6 +170,10 @@ async def _connect(self) -> aio_pika.RobustConnection:
         last_exception = None
         for _ in range(self._connection_attempts):
             try:
+
+                if not isinstance(self.port, int):
+                    raise RasaException("Port has to be a integer.")
```

Comment:
I think the RasaException can be raised a bit earlier - for example in the constructor [L75](https://github.com/RasaHQ/rasa/blob/a0399622910cc2ed7d003553fc664a3489a48ecc/rasa/core/brokers/pika.py#L75) where you can wrap the integer casting in a try-except block:
```
try:
     self.port = int(port)
except ValueError as e:
     raise RasaException("Port could not be converted to integer.") from e
```
You can try this by adding a new unit test at `rasa/tests/core/test_broker.py` and reproducing the Sentry error behaviour to see if this try-except block catches it.

Reply:
ok, thanks for the tips, would it be something like this?

Reply:
That was fast ⚡ 
It's looking in the right direction - I would also add an extra assertion after pytest.raises:
``` 
with pytest.raises(RasaException) as e:
     # to do

assert "Port could not be converted to integer." in str(e.value)
```